internal/armada/server/submit_to_log.go

```diff
@@ -330,6 +337,101 @@ func (srv *PulsarSubmitServer) CancelJobs(ctx context.Context, req *api.JobCance
 	}, nil
 }
 
+func (srv *PulsarSubmitServer) cancelJobsByIds(ctx context.Context, jobIds []string) (*api.CancellationResult, error) {
+	batchSize := srv.SubmitServer.cancelJobsBatchSize
+	batches := util.Batch(jobIds, batchSize)
+	idQueueJobSetMap := make(map[string]map[string][]string)
+
+	for _, batch := range batches {
+		jobs, err := srv.SubmitServer.jobRepository.GetJobsByIds(batch)
+		if err != nil {
+			return nil, status.Error(codes.Internal, "failed to get jobs from redis")
+		}
+		for _, job := range jobs {
+			q := job.Job.GetQueue()
+			jobSet := job.Job.GetJobSetId()
+			_, ok := idQueueJobSetMap[q]
+			if !ok {
+				idQueueJobSetMap[q] = make(map[string][]string)
+			}
+			_, ok = idQueueJobSetMap[q][jobSet]
+			if !ok {
+				idQueueJobSetMap[q][jobSet] = []string{}
+			}
+			idQueueJobSetMap[q][jobSet] = append(idQueueJobSetMap[q][jobSet], job.Job.Id)
+		}
+	}
+
+	var cancelledIds []string
+	var eventSequences []*armadaevents.EventSequence
+	for q, jobSets := range idQueueJobSetMap {
+		userId, groups, err := srv.Authorize(ctx, q, permissions.CancelAnyJobs, queue.PermissionVerbCancel)
+		if err != nil {
+			log.WithError(err).Errorf("user cannot cancel jobs in queue %s", q)
```

Comment:
Probably we should just return the error back to the user here and not do anything